### PR TITLE
Feature/fix windows and modify

### PR DESCRIPTION
**Pull request by sjeandeaux**
This PR contains:
* fix on windows URI
* fix on modify

### CODE REVIEW

**Comment by sjeandeaux** [comment]
fix #11 fix #12

**Comment by sjeandeaux** [comment]
Hi @johnweldon @vetinari @liggitt, 

I would like to know if there is a chance to review the PR?


Sorry to disturb you.

**Comment by johnweldon** [comment]
Hi @sjeandeaux - thank you for this PR - I appreciate the issues, the tests, and the changes.

I have a couple requests; 

1. on the issues could you describe a little more completely what you expected to see and what you actually saw (for example in #12 "nothing happens" isn't really clear).  Maybe:
  - what your input values are
  - what you expected the output to be, and
  - what the actual output was
  - contextual information like Go version, platform, etc.
 
2. In the issues could you also provide some supporting documentation for the url schemes?  For example, I don't personally know about the `/c:/foo/bar/baz` file path format being valid... is that a path format that windows expects to work, and can you provide supporting documentation for that?

Thanks again for participating and providing feedback

File: ldif.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func (l *LDIF) parseEntry(lines []string) (entry *Entry, err error) {
 	}
 
 	if !strings.HasPrefix(lines[0], "dn:") {
-		return nil, errors.New("Missing dn:")
+		return nil, errors.New("missing 'dn:'")
```
Comment (johnweldon) [inline ldif.go]: Good catch on the capitalization; what about `missing 'dn':`? (colon outside the single quotes)

File: ldif_test.go [hunk 2]
```diff
@@ -173,7 +175,9 @@ func TestLDIFURL(t *testing.T) {
 	defer os.Remove(f.Name())
 	f.Write([]byte("TEST\n"))
 	f.Sync()
-	l, err := ldif.Parse("dn: uid=someone,dc=example,dc=org\ndescription:< file://" + f.Name() + "\n")
+	d := strings.TrimPrefix(filepath.ToSlash(f.Name()), "C:")
```
Comment (johnweldon) [inline ldif_test.go]: I appreciate the cleanup; this feels a little brittle with assuming the tempfile will be created in the `C:` volume on windows... what happens if the test runs on windows with a temp dir on the `D:` volume?  Probably not super important in the tests, but something to think about.

**Comment by johnweldon** [comment]
I removed the comment I made about `'dn:'` vs. `'dn':`.  In re-reading the code, I think as-it-is makes sense.

**Comment by sjeandeaux** [comment]
Hi,

1. I added more information on the issue #12.
2. I added comments on function toPath and the error in the issue #11.

**Comment by johnweldon** [comment]
@vetinari @liggitt - Objections to merging this?

**Pull request merged**